File: 4107-find-missing-elements/find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        # nums.sort()
        # a = nums[0]
        # b = nums[-1]
        # newlist = []
        # for i in range(a+1, b):
        #     if i in nums:
        #         continue
        #     else:
        #         newlist.append(i)
        
        # return newlist

        st = set(nums)
        mn = min(nums)
        mx = max(nums)
        return [x for x in range(mn+1, mx) if x not in nums]

File: 4107-find-missing-elements/find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        vector<int> missing;
        sort(nums.begin(), nums.end());
        int start = nums[0];
        int end = nums.back();

        for(int i=start+1; i<end; i++){
            if(binary_search(nums.begin(), nums.end(), i)){
                continue;
            }else{
                missing.push_back(i);
            }
        }

        return missing;
    }
};